src/main/java/org/openRealmOfStars/player/fleet/Fleet.java

```diff
@@ -1190,41 +1187,26 @@ public double getTotalFleetCapacity() {
   }
 
   /**
-   * Get scarable ship based on obsolete models or just bigger fleet capacity.
+   * Get scrapable ship based on obsolete models or just bigger fleet capacity.
    * @param obsoleteModels List of ship design which are obsolete.
    * @return Ship or null if nothing to scrap.
    */
-  public Ship getScrableShip(final ShipStat[] obsoleteModels) {
-    Ship shipObsolete = null;
-    Ship shipNotObsolete = null;
-    for (Ship ship : ships) {
-      boolean obsolete = false;
-      for (int i = 0; i < obsoleteModels.length; i++) {
-        if (obsoleteModels[i].getDesign().getName().equals(ship.getName())) {
-          obsolete = true;
-          break;
-        }
-      }
-      if (obsolete) {
-        if (shipObsolete == null && ship.getFleetCapacity() > 0) {
-          shipObsolete = ship;
-        } else if (shipObsolete != null
-            && ship.getFleetCapacity() > shipObsolete.getFleetCapacity()) {
-          shipObsolete = ship;
-        }
-      } else {
-        if (shipNotObsolete == null && ship.getFleetCapacity() > 0) {
-          shipNotObsolete = ship;
-        } else if (shipNotObsolete != null
-            && ship.getFleetCapacity() > shipNotObsolete.getFleetCapacity()) {
-          shipNotObsolete = ship;
-        }
-      }
-    }
-    if (shipObsolete != null) {
-      return shipObsolete;
+  public Ship getScrapableShip(final ShipStat[] obsoleteModels) {
+    final Comparator<Ship> cmprFleetCapDesc = (a, b) -> {
+      return Double.compare(b.getFleetCapacity(), a.getFleetCapacity());
+    };
+
+    var obsoleteShips = ships.stream()
+        .filter(ship -> ship.isObsolete(obsoleteModels))
+        .sorted(cmprFleetCapDesc);
+    var notObsoleteShips = ships.stream()
```

Comment:
Similar here it goes all not obsolete ships. If there would be fleet without no obsolete and all would be unarmed then all of them could be scrap it one at time.